### PR TITLE
Add live collab editing notes

### PR DESCRIPTION
**Pull request by bokuanT**
T3-52

@thamruicong 

**Description**

Use third party open source library hocus pocus, that build off yjs package to integrate with current lexical editor to enable concurrent live editing. 
image

As SOC IT has told me they wont allow us to change the reverse proxy forwarding settings anymore, I added a API gateway (since hocus pocus needs its own server as it conflicts with backend that uses socket.io). Backend now runs on port 3003, and API gateway on 3001 (since reverse proxy is currently configured to forward /api/... to port 3001 on VM and we can't change this now)

Most changes outside of ./packages/hocus_pocus_server, frontend and backend are to accommodate for new hocus_pocus_server and api_gateway, as well as new API traffic endpoint ports

**Other Information**

**Checklist**

- [X] My pull request has a descriptive title (not a vague title like `Update README.md`)

- [X] My pull request targets the `master` branch of the repository only

- [X] My commits follows these [commit message guidelines](https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53)

- [ ] I added tests for the changes I made (if applicable)

- [X] I added or updated documentation (if applicable)

- [X] I have ran the project and its tests locally and verified that there are no visible errors

- [X] Includes DB Migration
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested

### CODE REVIEW

**Comment by bokuanT** [comment]
NUS IT says they won't support editing the reverse proxy configs- need to add own nginx to forward hocus pocus connection from frontend_url/api/TO_DECIDE to hocus pocus server

**Comment by bokuanT** [comment]
@thamruicong I have not changed VM docker compose and ENV yet, let me know after review then I'll update before merge

File: .github/workflows/deploy-to-staging.yaml [hunk 1]
```diff
@@ -66,6 +66,59 @@ jobs:
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # This step uses [docker/metadata-action](https://github.com/docker/metadata-action#about) to extract tags and labels that will be applied to the specified image. The `id` "meta" allows the output of this step to be referenced in a subsequent step. The `images` value provides the base name for the tags and labels.
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
+        with:
+          images: ${{ env.REGISTRY }}/${{env.IMAGE_NAME}}/hocus_poocus_server
```
Comment (thamruicong) [inline .github/workflows/deploy-to-staging.yaml:87]: image name should sync with build command `docker compose -f docker-compose-ci.yml build hocus_pocus_server`

**Comment by thamruicong** [comment]
LGTM!

**Pull request merged**